sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2387,11 +2387,19 @@ class Dataset[T] private[sql](
    */
   def withColumn(colName: String, col: Column): DataFrame = withColumns(Seq(colName), Seq(col))
 
+  /**
+   * Returns a new Dataset by adding a column at specific position or replacing the
+   * existing column that has the same name.
+   */
+  def withColumn(colName: String, col: Column, pos: Int): DataFrame =
```

Comment:
Not sure. I would just simply select.

Reply:
If use select, for users, which means they should  adjust the order of the cols.

Reply:
Instead of numeric position it would be way more convenient to reference a position by existing column names; perhaps add `before_column` and `after_column`?

Reply:
Agree with @HyukjinKwon, seems `select` can do this.

Reply:
yeah, as @HyukjinKwon said, i just realize with `select`.
![image](https://user-images.githubusercontent.com/19250651/98440846-1fc6b980-2136-11eb-899d-e2c16e3ea965.png)

Reply:
what if you have 100 cols, and add new col in 50th position?
 you have to get `val cols:Array[String] =df.columns` and adjust the order of the cols and then use selectExpr(cols: _* ).
that's my point.